### DISCUSSION TITLE
How to migrate from akhilrex to AlfHou container

### DISCUSSION THREAD

**Discussion by rlewis187** [AlfHou/hammond#6] [General]
Hi all, I'm pretty new to containers and GitHub and whatnot so please bear with me. 

I have the akhilrex/Hammond container running that I pulled from Docker Hub but would very much like to migrate to this new fork that has allowed development to continue. 
I have no idea how to go about doing it though can any one help please?

Thanks for taking on the development BTW!

**Comment by alexanderfitu** [comment]
I have the same Q. I think the first thing is for the docs on the main page to update, but suspect it will involve swapping the 

`  docker run -d -p 3000:3000 --name=hammond akhilrex/hammond`

to 
`  docker run -d -p 3000:3000 --name=hammond akhilrex/alfhou` 

and the same in the docker.yaml etc.

**Reply by rlewis187** [reply]
Thats only if someone creates a container and uploads it to dockerhub, and the new image is compatible with the old config. 
Unless I'm mistaken (which is a real possibility).

**Reply by alexanderfitu** [reply]
> Thats only if someone creates a container and uploads it to dockerhub, and the new image is compatible with the old config. Unless I'm mistaken (which is a real possibility).

No, that makes absolute sense. My only direct experience with docker containers is running and configuring them.

**Comment by alexanderfitu** [comment]
In the same vein, can we identify a migration path for data in the akhilrex version to the AlfHou fork :)

**Comment by alexanderfitu** [comment]
The latest release allows you to pull this from docker as an image which is great!

I only have a handful of data, so will likely just re enter it manually.

 A though, I created a few feature requests on the old project, shall I recreate them under this fork?

Great job.

**Comment by AlfHou** [comment]
Hi! I'm didn't receive any notifications that there were discussions going on here, so I just saw them now. Sorry about that!

There was a bit of catch up work to get done after I forked the project. But the first release is now out. It should be as simple as changing the docker command `docker run -d -p 3000:3000 --name=hammond akhilrex/hammond` to `docker run -d -p 3000:3000 --name=hammond alfhou/hammond`. as @alexanderfitu mentiones above. Just make sure it's using the same volume. I'll try to write a bit about it in the Readme over the weekend.

> [...] I created a few feature requests on the old project, shall I recreate them under this fork?

It would be great if you recreate them here. If not, I'll probably take a look at the issues in akhilrex's repo to find issues and features. But I'll prioritize issues/feature requests from this repo.

**Comment by rlewis187** [comment]
I use Portainer to manage my containers, it was as easy as changing the image from:

- akhilrex/hammond to 
- alfhou/hammond

![image](https://user-images.githubusercontent.com/80640955/215256521-44a5c70b-840c-44f2-bc3b-620d8d5181f2.png)

in the container edit page, 
All other settings I left the same.
Then Redeploy>Replace. Everything has migrated perfectly.
Many Thanks!

**Reply by kaciker** [reply]
> I use Portainer to manage my containers, it was as easy as changing the image from:
> 
> * akhilrex/hammond to
> * alfhou/hammond
> 
> ![image](https://user-images.githubusercontent.com/80640955/215256521-44a5c70b-840c-44f2-bc3b-620d8d5181f2.png)
> 
> in the container edit page, All other settings I left the same. Then Redeploy>Replace. Everything has migrated perfectly. Many Thanks!

Thanks, now I've migrated my previous data to this and works fine

**Comment by sebortner** [comment]
Can confirm. I just exchanged _akhilrex_ to _alfhou_ in my docker compose stack in portainer, redeployed and everything worked seamlessly. All the data was preserved. Excellent work!

**Comment by alexanderfitu** [comment]
Yes, as above all works well :)

**Comment by RobbieLD** [comment]
I tried the above and it mostly seemed to work well however when I first tried it all volume and distance units didn't work. I'm not sure why but I poked around in the db and then refreshed the page a few times and they are working again so I guess it was a caching issue on the client side. So if anyone else has this issue just force the client to refresh it's cache and you should be ok.
